frontend/src/v5/ui/routes/dashboard/projects/federations/federationSettingsForm/federationSettingsForm.component.tsx

```diff
@@ -17,25 +17,27 @@
 
 import { FederationSettingsSchema } from '@/v5/validation/containerAndFederationSchemes/federationSchemes';
 import { FederationsActionsDispatchers } from '@/v5/services/actionsDispatchers';
-import { IFederation } from '@/v5/store/federations/federations.types';
+import { FederationsHooksSelectors } from '@/v5/services/selectorsHooks';
 import { SettingsForm } from '../../settingsForm/settingsForm.component';
 
 type FederationSettingsFormProps = {
-	open: boolean;
-	federation: IFederation;
-	onClose: () => void;
+	federationId: string;
+	onClickClose: () => void;
 };
 
 export const FederationSettingsForm = ({
-	federation,
+	federationId,
```

Comment:
why the change from `federation` to `federationId`?
(same for container in the respective component)

Reply:
A problem with this method is that the props supplied to it can't be mutated. So when the FederationsSettingsForm calls fetchViews and fetchSettings even though the federation is updated in redux it wasn't being updated in the modal prop.

I'm starting to feel like this method might be good for some applications but not others, as it does carry with it a lot of potential to create bugs